### ISSUE TITLE
Allow grouping up component methods

### ISSUE THREAD

**Issue by ColinTree** [mit-cml/appinventor-sources#1727]
Just like properties, methods should be also allowed to be collapsed, and decided if they are going to br shown in the component method list. (Just like Clock.AddDays etc.)

**Comment by ewpatton** [comment]
This was implemented as more of a hack than anything else, and mostly because the signatures of those methods are all the same does it work. Do you have a proposal for how to generalize this? I assume that this would mainly be used by extensions? Or are there other places in core App Inventor that you see this being used?

**Comment by ColinTree** [comment]
Hi @ewpatton 

* I think this is necessary for all components, as you can see, drop down on Clock methods is even just a hack (totally not elegant).
* And there are a large amount of methods with similar functionality can be collapsed if there is such a functionality. Although those existing blocks may seems fine now, they would be better if they are not taking such a lot of space to be listed.
* Also, this can be a kind of replacement solution of dynamic arguments.
Dynamic arguments is actually the cue lead me to think about folding methods. Unlike Java or any other languages, App Inventor don't allow any methods with duplicated names (but with different combination of parameters). If there are any options on an operation, it must be implemented as a parameter or another method block. (e.g. `Canvas.DrawCircle(... fill)`; `Web.PostText()` & `Web.PutText()`)

**Comment by ewpatton** [comment]
Right, but the Clock implementation hack only really works because the arity of the methods is the same. Changing the method is really just changing the units involved in the computation. Method overloading was also something that I believe was explicitly excluded from the language for the purposes of discoverability and learning. If the methods have multiple potential meanings depending on the inputs, it is harder to reason about an app than if everything is explicit.

I'd like to see a more complete design doc with use cases of how this would work before we proceed. I would also like to get a sense from the education team about how it might affect teaching App Inventor.
